File: 백준/Silver/10773. 제로/제로.py
k = int(input())

a = []

for i in range(k):
    b = int(input())
    if b == 0:
        if a:  # 리스트가 비어있지 않으면
            a.pop()  # 마지막 요소 제거
    else:
        a.append(b)

c = sum(a)  # 리스트의 합 계산

print(c)

File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);

    int sum = 0;
    int n;
    stack<int> s;

    cin >> n;
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        if (num == 0) {
            s.pop();
        } else {
            s.push(num);
        }
    }
    n = s.size();
    for (int i = 0; i < n; i++) {
        sum += s.top();
        s.pop();
    }

    cout << sum;

    return 0;
}
    